source/MaterialXRender/OpenGL/GlslProgram.cpp

```diff
@@ -512,138 +509,31 @@ void GlslProgram::unbindGeometry()
     checkErrors();
 }
 
-void GlslProgram::unbindTextures()
+void GlslProgram::unbindTextures(ImageHandlerPtr imageHandler)
 {
-    for (GLint i=0; i<_textureUnitsInUse; i++)
-    { 
-        // Unbind a texture to that unit
-        glActiveTexture(GL_TEXTURE0 + i);
-        glBindTexture(GL_TEXTURE_2D, MaterialX::GlslProgram::UNDEFINED_OPENGL_RESOURCE_ID);
-        checkErrors();
-    }
-    _textureUnitsInUse = 0;
-
-    // Delete any allocated textures
-    if (_dummyTexture != MaterialX::GlslProgram::UNDEFINED_OPENGL_RESOURCE_ID)
-    {
-        glDeleteTextures(1, &_dummyTexture);
-        _dummyTexture = MaterialX::GlslProgram::UNDEFINED_OPENGL_RESOURCE_ID;
-    }
-    for (auto iter : _programTextures)
-    {
-        if (iter.second != MaterialX::GlslProgram::UNDEFINED_OPENGL_RESOURCE_ID)
-        {
-            glDeleteTextures(1, &iter.second);
-        }
-    }
-    _programTextures.clear();
-
+    imageHandler->clearImageCache();
```

Comment:
Texture handling code has been removed from here where it never really belonged.